### PR TITLE
add spark 4.0 preview news

### PR DESCRIPTION
**Pull request by cloud-fan**
Include 4.0.0-preview1 release in the documentation.md, add news post to announce Spark 4.0.0 preview1 release.

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @dongjoon-hyun @yaooqinn @gatorsmile

**Review by viirya — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @cloud-fan .

**Comment by cloud-fan** [comment]
thanks for the review!

File: news/_posts/2024-06-03-spark-4.0.0-preview1.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+either the mailing lists or JIRA.
+
+There are a lot of exciting new features added to Spark 4.0, including ANSI mode by default, Python data source, polymorphic
+Python UDTF, string collation support, new VARIANT data type, streaming state store data source, structured logging, Java 17 by default,
+and many more.
+
+We'd like to thank our contributors and users for their contributions and early feedback to this release. This release would not have
+been possible without you.
+
+To download Spark 4.0.0-preview1, head over to the <a href="https://archive.apache.org/dist/spark/spark-4.0.0-preview1">download page</a>.
```
Comment (yaooqinn) [inline news/_posts/2024-06-03-spark-4.0.0-preview1.md:26]: Is it deliberate to skip the docker release for 4.0.0-preview1?

Comment (cloud-fan) [reply news/_posts/2024-06-03-spark-4.0.0-preview1.md:26]: In the last preview release, there was no docker release, so there is no precedent we can follow. I'm not sure if we need to do it or not, cc @dongjoon-hyun @Yikun 

It looks like just some manual steps in the spark-docker repo and docker hub, shouldn't be hard.

Comment (yaooqinn) [reply news/_posts/2024-06-03-spark-4.0.0-preview1.md:26]: IMO, this step shouldn't be skipped and shall be added to the release guide if absent.

Comment (cloud-fan) [reply news/_posts/2024-06-03-spark-4.0.0-preview1.md:26]: @Yikun can you help with it? I'm busy with the data+ai summit these days and can do it next week.

**Pull request merged**